src/sentry/services/hybrid_cloud/filter_query.py

```diff
@@ -75,7 +75,7 @@ def serialize_rpc(self, object: BASE_MODEL) -> RPC_RESPONSE:
     def _filter_has_any_key_validator(self, *keys: str) -> Callable[[FILTER_ARGS], Optional[str]]:
         def validator(d: FILTER_ARGS) -> Optional[str]:
             for k in keys:
-                if k in d:  # type: ignore # We assume FILTER_ARGS is a dict
+                if k in d:  # type: ignore[operator]  # We assume FILTER_ARGS is a dict
```

Comment:
This is giving me trouble. We can't just write `bound=TypedDict` because

```
src/sentry/services/hybrid_cloud/filter_query.py:18: error: Variable "typing_extensions.TypedDict" is not valid as a type  [valid-type]
src/sentry/services/hybrid_cloud/filter_query.py:18: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
```

But using an actual type alias such as `bound=Dict[str, Any]` doesn't work either:

```
src/sentry/services/hybrid_cloud/identity/impl.py:93: error: Type argument "IdentityFilterArgs" of "FilterQueryDatabaseImpl" must be a subtype of "Dict[str, Any]"  [type-var]
src/sentry/services/hybrid_cloud/user_option/impl.py:56: error: Type argument "UserOptionFilterArgs" of "FilterQueryDatabaseImpl" must be a subtype of "Dict[str, Any]"  [type-var]
src/sentry/services/hybrid_cloud/user/impl.py:156: error: Type argument "UserFilterArgs" of "FilterQueryDatabaseImpl" must be a subtype of "Dict[str, Any]"  [type-var]
[etc.]
```

I experimented with defining a new intermediate class like

```python
class QueryArgs(TypedDict, total=False):
    pass

FILTER_ARGS = TypeVar("FILTER_ARGS", bound=QueryArgs)
```

which makes the parameterized annotations happy but defeats our original intent of using `TypedDict`, which is that we want to initialize only some of the attributes and have a missing key mean "don't filter by this" (in contrast to a `None` value, which means "filter by the value `None` in this field"). There is a very confusing error:

> src/sentry/services/hybrid_cloud/user/impl.py:141: error: Missing keys ("is_active", "organization_id", "emails", "email_verified", "query", "authenticator_types") for TypedDict "UserFilterArgs"  [typeddict-item]

This message (especially "[typeddict-item]") suggests that our approach ("it's okay to initialize partially, and missing keys carry semantic meaning") is wrong, or at least inconsistent Mypy's opinions about `TypedDict`s. @asottile-sentry Can you offer any insight here?

Reply:
I think the `FILTER_ARGS` is maybe just wrong?  should it be `d: Container[str]` ?